admin/class-manage-fonts.php

```diff
@@ -296,7 +305,82 @@ function ( $font_family ) use ( $font_slug ) {
 			get_stylesheet_directory() . '/theme.json',
 			$theme_json_string
 		);
+	}
+
+	function manage_font_license( $font_name, $file_name ) {
+		if ( ! $font_name ) {
+			return;
+		}
+
+		// Get theme readme.txt
+		$readme_file          = get_stylesheet_directory() . '/readme.txt';
+		$readme_file_contents = file_get_contents( $readme_file );
+
+		if ( ! $readme_file ) {
+			return;
+		}
+
+		// Format font name for font source link
+		$google_formatted_font_name = str_replace( ' ', '+', $font_name );
+
+		// If file_name exists, then add font license to readme.txt
+		if ( 'remove' !== $file_name && is_string( $file_name ) ) {
+			// Check that the font is not already credited in readme.txt
+			if ( false === stripos( $readme_file_contents, $font_name ) ) {
+				// Require php-font-lib
+				require_once( __DIR__ . '/../includes/FontLib/Autoloader.php' );
+
+				// Build font source URL
+				$font_source = 'https://www.google.com/fonts/specimen/' . $google_formatted_font_name;
```

Comment:
Could we get rid of this and use the info from the font file asset metadata? In this way we could use this function for local font files too.

Reply:
Yes good idea, I was hoping to extend this to local fonts once I got this version working ✨